push/push.go

```diff
@@ -101,6 +101,13 @@ func GithubPush(ctx context.Context, input Input, repoLimiter *time.Ticker, push
 	tc := oauth2.NewClient(ctx, ts)
 	client := github.NewClient(tc)
 
+	if os.Getenv("GITHUB_URL") != "" {
+		baseEndpoint, _ := url.Parse(os.Getenv("GITHUB_URL"))
+		client.BaseURL = baseEndpoint
+		uploadEndpoint, _ := url.Parse(os.Getenv("GITHUB_URL") + "upload/")
+		client.UploadURL = uploadEndpoint
+	}
```

Comment:
by "rule of 3" would love if we could refactor this logic (perhaps all of `createGithubClient`) into a helper fn so it's not repeated throughout

Reply:
Yes, I was thinking this as I was adding the same code in all these different places.